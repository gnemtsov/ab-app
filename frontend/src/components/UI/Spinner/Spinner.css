.Loader {
  position: relative;
  width: 4rem;
  height: 4rem;
  margin: 8rem auto;
}

.Loader div {
  position: absolute;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background: #0a1372;
  animation: Loader 1.2s linear infinite;
}
.Loader div:nth-child(1) {
  top: 6px;
  left: 6px;
  animation-delay: 0s;
}
.Loader div:nth-child(2) {
  top: 6px;
  left: 26px;
  animation-delay: -0.4s;
}
.Loader div:nth-child(3) {
  top: 6px;
  left: 45px;
  animation-delay: -0.8s;
}
.Loader div:nth-child(4) {
  top: 26px;
  left: 6px;
  animation-delay: -0.4s;
}
.Loader div:nth-child(5) {
  top: 26px;
  left: 26px;
  animation-delay: -0.8s;
}
.Loader div:nth-child(6) {
  top: 26px;
  left: 45px;
  animation-delay: -1.2s;
}
.Loader div:nth-child(7) {
  top: 45px;
  left: 6px;
  animation-delay: -0.8s;
}
.Loader div:nth-child(8) {
  top: 45px;
  left: 26px;
  animation-delay: -1.2s;
}
.Loader div:nth-child(9) {
  top: 45px;
  left: 45px;
  animation-delay: -1.6s;
}
@keyframes Loader {
  0%, 100% {
    opacity: 1;
  }
  50% {
    opacity: 0.2;
  }
}
