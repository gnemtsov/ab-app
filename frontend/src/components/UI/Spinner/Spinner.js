import React from 'react';

import classes from './Spinner.css';

const spinner = () => (
    <div className={classes.Loader}><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div>
);

export default spinner;