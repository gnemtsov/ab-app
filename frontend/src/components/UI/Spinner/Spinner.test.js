import React from 'react';

import { configure, render, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Spinner from './Spinner';

configure({ adapter: new Adapter() });

describe('Spinner', () => {
	it('Renders correctly', () => {
		const wrapper = shallow(<Spinner />);
		expect(wrapper).toMatchSnapshot();
	});
})