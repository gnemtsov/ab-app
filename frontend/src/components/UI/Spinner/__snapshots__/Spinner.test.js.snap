// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Renders correctly 1`] = `
<div
  className="Loader"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
